src/components/ShoppingListItem/ShoppingListItem.js

```diff
@@ -2,7 +2,12 @@ import { useState, useEffect } from 'react';
 
 import isUnder24hSincePurchased from '../../utils/isUnder24hSincePurchased.js';
 
-const ShoppingListItem = ({ listId, itemId, item, checkAsPurchased }) => {
+const ShoppingListItem = ({
+  itemId,
+  item,
+  checkAsPurchased,
+  handleModalOpen,
```

Comment:
Passing this down so many levels makes me uncomfortable, but since the app if functionally finished at this point, I think it's fine.

Reply:
This was definitely a struggle! We originally had most of this modal code right at ShoppingList. But because we add an `aria-hidden=true` to hide the App when the modal is open, we had to move everything up to App.js. We can definitely  consider refactoring to use React Context in the future so we don't have to prop drill something like this so many levels down.

Reply:
@nick-zanetti Maybe we can make an issue for refactoring some things into React context?

Reply:
@anderswift @connietran-dev I actually think that would be more work than it's worth, since the app is now essentially done functionally.

Reply:
@nick-zanetti I don't think it's necessary, but might be a good opportunity to practice with React context if anyone has some time and wants to give it a try. Refactoring can be a really great learning experience.

Reply:
I've logged #45 as a future enhancement/issue we could take on 😃